packages/api-docs-builder/buildApiUtils.ts

```diff
@@ -520,3 +564,172 @@ export const stringifySymbol = (symbol: ts.Symbol, project: TypeScriptProject) =
+          const componentNameKebabCase = kebabCase(component);
+          demosTabImportStatements += `import ${component}ApiJsonPageContent from '../api/${componentNameKebabCase}.json';`;
+          apiTabImportStatements += `import ${component}ApiJsonPageContent from '../../api/${componentNameKebabCase}.json';`;
+        });
+      }
+
+      if (hooks) {
+        hooks.forEach((hook: string) => {
+          const hookNameKebabCase = kebabCase(hook);
+          demosTabImportStatements += `import ${hook}ApiJsonPageContent from '../api/${hookNameKebabCase}.json';`;
+          apiTabImportStatements += `import ${hook}ApiJsonPageContent from '../../api/${hookNameKebabCase}.json';`;
+        });
+      }
+
+      let staticProps = `
+      export const getStaticProps = () => {
+      `;
+
+      if (components) {
+        components.forEach((component: string) => {
+          const componentNameKebabCase = kebabCase(component);
+          staticProps += `
+          const ${component}ApiReq = require.context(
+            'docs/translations/api-docs/${componentNameKebabCase}',
+            false,
+            /${componentNameKebabCase}.*.json$/,
+          );
+          const ${component}ApiDescriptions = mapApiPageTranslations(${component}ApiReq);
+          `;
+        });
+      }
+
+      if (hooks) {
+        hooks.forEach((hook: string) => {
+          const hookNameKebabCase = kebabCase(hook);
+          staticProps += `
+          const ${hook}ApiReq = require.context(
+            'docs/translations/api-docs/${hookNameKebabCase}',
+            false,
+            /${hookNameKebabCase}.*.json$/,
+          );
+          const ${hook}ApiDescriptions = mapApiPageTranslations(${hook}ApiReq);
+          `;
+        });
+      }
+
+      staticProps += `
+        return {
+          props: {
+            componentsApiDescriptions: { `;
+
+      if (components) {
+        components.forEach((component: string) => {
+          staticProps += `${component} : ${component}ApiDescriptions ,`;
+        });
+      }
+
+      staticProps += `},
+          componentsApiPageContents: { `;
+
+      if (components) {
+        components.forEach((component: string) => {
+          staticProps += `${component} : ${component}ApiJsonPageContent ,`;
+        });
+      }
+
+      staticProps += ` },
+          hooksApiDescriptions: { `;
+
+      if (hooks) {
+        hooks.forEach((hook: string) => {
+          staticProps += `${hook} : ${hook}ApiDescriptions ,`;
+        });
+      }
+
+      staticProps += ` },
+          hooksApiPageContents: { `;
+
+      if (hooks) {
+        hooks.forEach((hook: string) => {
+          staticProps += `${hook} : ${hook}ApiJsonPageContent ,`;
+        });
+      }
+
+      staticProps += ` },},};};`;
+
+      const tokens = markdown.pathname.split('/');
+      const name = tokens[tokens.length - 1];
+      const importStatement = `docs/data${markdown.pathname}/${name}.md`;
+      const demosSource = `
```

Comment:
This page doesn't necessarily need to be generated, it can be copied as previously when adding the demos page, but I added this step to ensure that the `getLayout` method would be used, as it is not there on all pages at this moment. Once we migrate all pages to the new structure, we can remove this step.